packages/@aws-cdk/aws-ec2/lib/user-data.ts

```diff
@@ -131,6 +131,21 @@ export abstract class UserData {
    */
   public abstract addSignalOnExitCommand( resource: Resource ): void;
 
+
+  /**
+   * Use this to set whether or not this userdata will persist through
+   * an instance reboot; allowing it to run on every instance start.
+   * By default, UserData is run only once during the first instance launch.
+   *
+   * Only available on WindowsUserData.
+   *
+   * For more information, see:
+   * https://aws.amazon.com/premiumsupport/knowledge-center/execute-user-data-ec2/
+   * https://docs.aws.amazon.com/AWSEC2/latest/WindowsGuide/ec2-windows-user-data.html#user-data-scripts
+   *
+   * @param value If true then the userdata is persisted. (Windows only)
+   */
+  public abstract persist(value: boolean): void;
```

Comment:
Given that half of the locations of this function throw errors when called, I don't think this is the right level of abstraction. I'm also not sure about this contract as a whole. Since this is a function, not a field, calling it implies that the user wants their data to persist. Given that, it seem that the boolean should be unnecessary.

Reply:
I completely agree, however the CDK does not export the internal `LinuxUserData` or `WindowsUserData` classes. I originally just wanted to add the property to `WindowsUserData`, but then of course a reference to `UserData` trying to touch that property fails type checks. Without `WindowsUserData` being exported it's not even possible to do `instance.userData as WindowsUserData`

So, we have two options.
1. Accept that the property is on the base `UserData` class and only makes sense for one of the options; or
2. Export at least `WindowsUserData` so that users can type-cast to be able to set `persist` on it.

I went with (1), obviously. What would the CDK team prefer?

Reply:
@TheRealAmazonKendra  Before the bot kills this PR. Any response to my question?

Reply:
@TheRealAmazonKendra Still waiting on your thoughts? Should I just add the integ test to the current PR's form, or switch to exporting `WindowsUserData` and only having the new `persist` method on `WindowsUserData` instead of all UserData classes?

Reply:
I think the best option is actually to add this as an option in `forWindows`. `forLinux` has an option added it that is only specific to it so I think duplicating that experience makes sense in this context.

Reply:
The suggestion has been implemented.